src/averageAges.js

```diff
@@ -55,7 +74,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(person => person.sex === 'm'
+    && people.some(mother => mother.name === person.mother))
+    : people.filter(person => (
+      people.some(mother => mother.name === person.mother)));
+
+  const ageBetween = children.map(child => (
+    child.born - people.find(mother => mother.name === child.mother).born));
+
+  return ageBetween.reduce((prev, current) => (
+    prev + current), 0) / ageBetween.length;
```

Comment:
you use identical reduce three times, it is better to create a separate function with this reduce and use this function